agent/api/task/service_connect.go

```diff
@@ -65,3 +81,246 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ParseServiceConnectAttachment parses the service connect container name and service connect config value
+// from the passed attachment.
+func ParseServiceConnectAttachment(scAttachment *ecsacs.Attachment, taskContainers []*ecsacs.Container, networkMode string) (*ServiceConnectConfig, error) {
+	scConfigValue := &ServiceConnectConfig{}
+	containerName := ""
+	for _, property := range scAttachment.AttachmentProperties {
+		switch aws.StringValue(property.Name) {
+		case serviceConnectConfigKey:
+			// extract service connect config value from attachment properties,
+			// and translate ecsacs.Attachment.Value to ServiceConnectConfig
+			data, err := strconv.Unquote(string(*property.Value))
+			if err != nil {
+				return nil, errors.Errorf("failed to unqote attachment property value which contains service connect config value: %s", err)
+			}
+			if err := json.Unmarshal([]byte(data), scConfigValue); err != nil {
+				return nil, errors.Errorf("failed to unmarshal attachment property value which contains service connect config value: %s", err)
+			}
+		case serviceConnectContainerNameKey:
+			// extract service connect container name from attachment properties
+			containerName = aws.StringValue(property.Value)
+		default:
+			return nil, errors.Errorf("received an unrecognized attachment property: %s", property.String())
+		}
+	}
+	scConfigValue.ContainerName = containerName
+	if err := ValidateSCConfig(scConfigValue, taskContainers, networkMode); err != nil {
+		return nil, errors.Errorf("service connect config validation failed: %s", err)
```

Comment:
nit: the preferred way to wrap errors is now `fmt.Errorf("message %w", err)`

Reply:
Updates are made in commit 94753650e02342a2c1be58645e8e187489efe452. Thanks!